renku/ui/service/serializers/datasets.py

```diff
@@ -131,7 +132,7 @@ class DatasetDetailsResponse(DatasetDetails):
     images = fields.List(fields.Nested(ImageObject))
 
 
-class DatasetListResponse(Schema):
+class DatasetListResponse(GitUrlResponseMixin):
```

Comment:
Adding mixins and using inheritance in all the schemas like this makes the code really hard to read and follow imo. To see what fields the `DatasetListResponse` contains you have to click through several levels of inheritance until you find a specific field. And there is no single place where you can look at to see what the full payload is supposed to be.

I know this is not something new but I think we should try to use it a lot more sparingly.

Reply:
For example for `ManifestTemplatesRequest_v2_2` in `renku/ui/service/serializers/v1/templates.py` I was wondering why we have these lines in the `pre_load` section of the schema (because they do not apper in the code in `ManifestTemplatesRequest_v2_2`):

```
data["git_url"] = data.get("url")
data["template_git_url"] = data["git_url"]
```

So to find them I had to step through:
- `ManifestTemplatesRequest_v2_2`
- `RepositoryCloneRequest`
- `RemoteRepositorySchema`
- `RemoteRepositoryBaseSchema`

I think we should stick to just 1 level of inheritance or 2 levels in extreme situations. And if we need to reuse things we should reuse/compose field definitions that are common across different schemas rather than have this many levels of inheritance.

And even after all this I cannot see where we need `template_git_url`

Reply:
We should not try to fix this now. But perhaps it is something we can consider simplifying in the cooldown period.

Reply:
Ok I found `template_git_url` it seems I was not at the end of the inheritance chain at `ManifestTemplatesRequest_v2_2` - as I assumed.

The whole chain is:
- `ProjectTemplateRequest_v2_2`
- `ManifestTemplatesRequest_v2_2`
- `RepositoryCloneRequest`
- `RemoteRepositorySchema`
- `RemoteRepositoryBaseSchema`

I am sorry for the long comments - I just wanted to illustrate my point.

Reply:
I completely agree, as it currently is, it's terrible. There's also some endpoints that take parameters that they don't use, just because a schema was inherited that doesn't fully match. I removed some of that here, where the template endpoints took project clone values that they just ignored for no good reason.
I think cooldown is a good fit for addressing this